NumberFormatException in zxing when generating QR-Code

`java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3114)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3257)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1948)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
Caused by: java.lang.NumberFormatException: 
  at java.lang.Integer.parseInt (Integer.java:615)
  at java.lang.Integer.parseInt (Integer.java:650)
  at com.google.zxing.aztec.AztecWriter.encode (AztecWriter.java:51)
  at com.google.zxing.MultiFormatWriter.encode (MultiFormatWriter.java:102)
  at com.secuso.privacyfriendlycodescanner.qrscanner.helpers.Utils.generateCode (Utils.java:43)
  at com.secuso.privacyfriendlycodescanner.qrscanner.helpers.Utils.generateCode (Utils.java:37)
  at com.secuso.privacyfriendlycodescanner.qrscanner.ui.viewmodel.ResultViewModel.createHistoryItem (ResultViewModel.java:102)
  at com.secuso.privacyfriendlycodescanner.qrscanner.ui.viewmodel.ResultViewModel.initFromScan (ResultViewModel.java:69)
  at com.secuso.privacyfriendlycodescanner.qrscanner.ui.activities.ResultActivity.initStateIfNecessary (ResultActivity.java:111)
  at com.secuso.privacyfriendlycodescanner.qrscanner.ui.activities.ResultActivity.onCreate (ResultActivity.java:81)
  at android.app.Activity.performCreate (Activity.java:7327)
  at android.app.Activity.performCreate (Activity.java:7318)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1271)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3094)`

--- a/app/src/main/java/com/secuso/privacyFriendlyCodeScanner/qrscanner/helpers/Utils.java
+++ b/app/src/main/java/com/secuso/privacyFriendlyCodeScanner/qrscanner/helpers/Utils.java
@@ -30,7 +30,7 @@ public class Utils {
                 hints.put(ERROR_CORRECTION, ec);
             }
         }
-        if(!hints.containsKey(ERROR_CORRECTION)) {
+        if(!hints.containsKey(ERROR_CORRECTION) && format!=BarcodeFormat.AZTEC) {
             hints.put(ERROR_CORRECTION, ErrorCorrectionLevel.L.name());
         }
 

